capreolus/benchmark/msmarco.py

```diff
@@ -1,14 +1,108 @@
-from capreolus import constants
+import os
+import math
+import gzip
+import json
+import random
+
+from capreolus import constants, ConfigOption, Dependency, constants
+from capreolus.utils.common import download_file, remove_newline
+from capreolus.utils.trec import topic_to_trectxt, load_trec_topics
+from capreolus.utils.loginit import get_logger
+
+logger = get_logger(__name__)
 
 PACKAGE_PATH = constants["PACKAGE_PATH"]
+from . import Benchmark
+
+
+@Benchmark.register
+class MSMarcoPassage(Benchmark):
+    module_name = "msmarcopsg"
+    dependencies = [Dependency(key="collection", module="collection", name="msmarcopsg")]
+
+    query_type = "title"
+    config_spec = []
+    use_train_as_dev = False
+
+    data_dir = PACKAGE_PATH / "data" / "msmarcopsg"
+    qrel_file = data_dir / "qrels.txt"
+    topic_file = data_dir / "topics.txt"
+    fold_file = data_dir / "folds.json"
+
+    def build(self):
+        self.download_if_missing()
+
+    def download_if_missing(self):
+        self.data_dir.mkdir(exist_ok=True, parents=True)
+        if all([f.exists() for f in [self.qrel_file, self.topic_file, self.fold_file]]):
+            return
+
+        def match_size(fn):
+            if ".train." in fn:
+                return True
+
+            # if self.config["qrelsize"] == "small":
+            if True:
+                return ".small." in fn
+            return ".small." not in fn
+
+        gz_dir = self.collection.download_raw()
+        queries_fn = [fn for fn in os.listdir(gz_dir) if "queries." in fn and match_size(fn)]
+        qrels_fn = [fn for fn in os.listdir(gz_dir) if "qrels." in fn and match_size(fn)]  # note that qrel.test is not given
+
+        # topic and qrel
+        topic_f, qrel_f = open(self.topic_file, "w"), open(self.qrel_file, "w")
+        folds = {"train": set(), "dev": set(), "eval": set()}
+
+        for set_name in folds:
```

Comment:
So if I understood it correctly, we download the raw files to a temp folder and then we copy it over to our msmarco collection's `data_dir`. This would allow us to create folds e.t.c from the raw msmarco data (though we have only 1 fold now). Did I understand that correctly?

Reply:
yup exactly. they will be stored under `data_dir/msmarcopsg`